### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by neharajesh**

### CODE REVIEW

File: components/input/input.html [hunk 1]
```diff
@@ -75,21 +75,44 @@ <h1 id="header-name">purplecss</h1>
...
+            <p>Outlined textfields where border appears on focus.</p>
+            <input placeholder="Basic" type="text" class="textfield textfield-basic">
+            <input placeholder="Background" type="text" class="textfield textfield-bg">
+            <input placeholder="Outline" type="text" class="textfield textfield-outline">
+            <iframe
+            src="https://carbon.now.sh/embed?bg=rgba%28255%2C255%2C255%2C1%29&t=shades-of-purple&wt=none&l=htmlmixed&ds=false&dsyoff=20px&dsblur=68px&wc=true&wa=false&pv=17px&ph=21px&ln=false&fl=1&fm=Hack&fs=14px&lh=143%25&si=false&es=1x&wm=false&code=%253Cinput%2520placeholder%253D%2522Basic%2522%2520type%253D%2522text%2522%2520class%253D%2522textfield%2520textfield-basic%2522%253E%250A%253Cinput%2520placeholder%253D%2522Background%2522%2520type%253D%2522text%2522%2520class%253D%2522textfield%2520textfield-bg%2522%253E%250A%253Cinput%2520placeholder%253D%2522Outline%2522%2520type%253D%2522text%2522%2520class%253D%2522textfield%2520textfield-outline%2522%253E"
+            style="width: 904px; height: 200px; border:0; transform: scale(1); overflow:hidden;"
+            sandbox="allow-scripts allow-same-origin">
```
Comment (DanishOnGit) [inline components/input/input.html:88]: Hi @neharajesh ,
we should avoid inline styles as much as possible. You can transfer them to an external sheet

File: components/input/input.html [hunk 2]
```diff
@@ -75,21 +75,44 @@ <h1 id="header-name">purplecss</h1>
...
+            src="https://carbon.now.sh/embed?bg=rgba%28255%2C255%2C255%2C1%29&t=shades-of-purple&wt=none&l=htmlmixed&ds=false&dsyoff=20px&dsblur=68px&wc=true&wa=false&pv=17px&ph=21px&ln=false&fl=1&fm=Hack&fs=14px&lh=143%25&si=false&es=1x&wm=false&code=%253Clabel%2520class%253D%2522textfield-label%2520label-top%2522%2520for%253D%2522name%2522%253EName%253C%252Flabel%253E%250A%253Cinput%2520type%253D%2522text%2522%2520class%253D%2522textfield%2520textfield-basic%2522%2520id%253D%2522name%2522%253E"
+            style="width: 904px; height: 148px; border:0; transform: scale(1); overflow:hidden;"
+            sandbox="allow-scripts allow-same-origin">
+            </iframe>
+
+            <br>
+            <p>Textfield with Label on the left.</p>
+            <div class="textfield-container">
+                <label class="textfield-label label-left" for="age">Number</label>
+                <input type="number" class="textfield textfield-bg" id="age" style="margin-top: 15px;">
```
Comment (MKumaran98) [inline components/input/input.html:109]: Avoiding inline styles is good practise

File: styles.css [hunk 3]
```diff
@@ -462,36 +462,67 @@ button {
...
+input {
+    outline: none;
 }
 
-.input-username > span {
-    background: gray;
+.textfield {
+    width: 150px;
+    height: 40px;
+    padding: 3px 8px;
```
Comment (DanishOnGit) [inline styles.css:472]: You can use relative units here

File: components/input/input.html [hunk 4]
```diff
@@ -75,21 +75,44 @@ <h1 id="header-name">purplecss</h1>
...
     <div class="content">
         <div class="container">
-            <h1>Input</h1>
-            <h2>Textbox</h2>
-            <!-- FIX! -->
-            <div class="input-textbox">
-                <span>Enter text here</span>
-                <textarea class="input-textarea" aria-label="enter text here"></textarea>
-            </div>
+            <h1 class="div-heading">Simple Textfields</h1>
```
Comment (DanishOnGit) [inline components/input/input.html:78]: Class name can be made better . For eg: container-heading or anything better...

File: components/input/input.html [hunk 5]
```diff
@@ -75,21 +75,44 @@ <h1 id="header-name">purplecss</h1>
...
-                    <input type="text" placeholder="Username"/>
-                </div>
-            </div>
+            <hr>
+
+            <h1 class="div-heading">Simple Textfields with Labels</h1>
+            <p>Textfield with Label on top.</p>
+            <div class="textfield-container">
+                <label class="textfield-label label-top" for="name">Name</label>
+                <input type="text" class="textfield textfield-basic" id="name">
```
Comment (MKumaran98) [inline components/input/input.html:97]: `<label class="textfield-label label-top">Name
    <input type="text" class="textfield textfield-basic">
</label>`

nesting the input tag inside the label will remove the need for id and for properties

**Review by DanishOnGit — COMMENTED** [review]
Hi @neharajesh ,
I have suggested about class names,inline-styles, and css units.

**Review by MKumaran98 — COMMENTED** [review]
Hey Neha,
The code looks good overall, I have added some comments do check them out.

File: components/input/input.html (hunk 5 above)
Comment (neharajesh) [reply components/input/input.html:97]: oh! i din't know that! thankyou!

**Pull request merged**